src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Field.cs

```diff
@@ -12,11 +11,73 @@ public override BoundNode VisitFieldAccess(BoundFieldAccess node)
+            }
+        }
+
+        /// <summary>
+        /// Converts access to a tuple instance into access into the underlying ValueTuple(s).
+        ///
+        /// For instance, tuple.Item8
+        /// produces fieldAccess(field=Item1, receiver=fieldAccess(field=Rest, receiver=ValueTuple for tuple))
+        /// </summary>
+        private BoundExpression RewriteTupleFieldAccess(BoundFieldAccess node, BoundExpression rewrittenReceiver)
+        {
+            var tupleField = (TupleFieldSymbol)node.FieldSymbol;
+            var tupleType = (TupleTypeSymbol)tupleField.ContainingSymbol;
+            NamedTypeSymbol underlyingTupleType = tupleType.UnderlyingTupleType;
+
+            int fieldRemainder;
+            int fieldChainLength = TupleTypeSymbol.NumberOfValueTuples(tupleField.Position, out fieldRemainder);
+
+            BoundExpression currentField = rewrittenReceiver;
+            NamedTypeSymbol currentLinkType = underlyingTupleType;
+
+            // make nested field accesses to Rest
+            for (int fieldLinkIndex = 0; fieldLinkIndex < fieldChainLength - 1; fieldLinkIndex++)
+            {
+                WellKnownMember wellKnownTypeMember = TupleTypeSymbol.GetTupleTypeMember(TupleTypeSymbol.chunk8, TupleTypeSymbol.chunk8);
+                var linkField = (FieldSymbol)Binder.GetWellKnownTypeMember(_compilation, wellKnownTypeMember, _diagnostics, syntax: node.Syntax);
+                if ((object)linkField == null)
+                {
+                    return node;
+                }
+                FieldSymbol nestedFieldSymbol = linkField.AsMember(currentLinkType);
+
+                TypeSymbol nestedFieldType = currentLinkType.TypeArguments[TupleTypeSymbol.chunk7];
+                currentField = new BoundFieldAccess(node.Syntax, currentField, nestedFieldSymbol, ConstantValue.NotAvailable, LookupResultKind.Viable, nestedFieldType);
+
+                currentLinkType = TupleTypeSymbol.GetNestedTupleType(currentLinkType, 1);
+            }
+
+            {
+                // make a field access for the most local access
+                int tupleRemainder;
+                int tupleChainLength = TupleTypeSymbol.NumberOfValueTuples(tupleType.Arity, out tupleRemainder);
+
+                int containingLinkSize = (tupleChainLength == fieldChainLength) ? tupleRemainder : TupleTypeSymbol.chunk8;
+
+                WellKnownMember wellKnownTypeMember = TupleTypeSymbol.GetTupleTypeMember(containingLinkSize, fieldRemainder);
+                var linkField = (FieldSymbol)Binder.GetWellKnownTypeMember(_compilation, wellKnownTypeMember, _diagnostics, syntax: node.Syntax);
+                if ((object)linkField == null)
+                {
+                    return node;
+                }
+
+                FieldSymbol nestedFieldSymbol = linkField.AsMember(currentLinkType);
+
+                TypeSymbol nestedFieldType = currentLinkType.TypeArguments[fieldRemainder - 1];
```

Comment:
Sure. This was an artifact of keeping the code inside the loop and outside the loop similar.